code/SiteConfig.php

```diff
@@ -271,11 +271,13 @@ public static function current_site_config()
     {
         /** @var SiteConfig $siteConfig */
         $siteConfig = DataObject::get_one(SiteConfig::class);
-        if ($siteConfig) {
-            return $siteConfig;
+        if (!$siteConfig) {
+            $siteConfig = self::make_site_config();
         }
 
-        return self::make_site_config();
+        $siteConfig->extend('updateCurrentSiteConfig', $siteConfig);
```

Comment:
`$siteConfig` calling extend and being an argument doesn't look right

You used `static::singleton()->extend(` for a static extension hook [here](https://github.com/silverstripe/silverstripe-framework/pull/10027/files#diff-983f05f9a107b9bf66a1ed4230e524be82118dd71f4a795529f8794b762c8b47R878) - would that also work here as well?

Reply:
Yeah, the `$siteConfig` argument isn't needed because it's accessible as `$this->owner` in extensions.

Reply:
Used `static::singleton()->extend` as I think it's easier to replace the config that way. Not sure how I'd be replacing this->owner with a new instance (not saying it's impossible), but using the arg via a reference feels cleaner.